app/main.py

```diff
@@ -1,8 +1,51 @@
 class Car:
-    # write your code here
-    pass
+
+    def __init__(
+            self,
+            comfort_class: int,
+            clean_mark: int,
+            brand: str) -> None:
+
+        self.comfort_class = comfort_class
+        self.clean_mark = clean_mark
+        self.brand = brand
 
 
 class CarWashStation:
-    # write your code here
-    pass
+
+    def __init__(
+            self,
+            distance_from_city_center: int,
+            clean_power: int,
+            average_rating: float,
+            count_of_ratings: int) -> None:
+
+        self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        self.average_rating = average_rating
+        self.count_of_ratings = count_of_ratings
+
+    def serve_cars(self, list_cars: list) -> float:
+        total = 0
+        for car in list_cars:
+            if car.clean_mark < self.clean_power:
+                total += self.calculate_washing_price(car)
+                self.wash_single_car(car)
+
+        return round(total, 1)
+
+    def calculate_washing_price(self, car: Car) -> float:
+        calc = car.comfort_class * (self.clean_power - car.clean_mark)
+        calc *= self.average_rating / self.distance_from_city_center
+
+        return round(calc, 1)
+
+    def wash_single_car(self, car: Car) -> None:
+        car.clean_mark = self.clean_power
+
+    def rate_service(self, taxa: int) -> None:
+        self.average_rating *= self.count_of_ratings
+        self.average_rating += taxa
+        self.average_rating /= self.count_of_ratings + 1
+        self.average_rating = round(self.average_rating, 1)
+        self.count_of_ratings += 1
```

Comment:
A conta está certa, mas a implementação tem um risco: ela modifica average_rating antes de atualizar count_of_ratings, o que pode causar erro se o código for interrompido entre essas operações.

```
def rate_service(self, taxa: int) -> None:
    total_rating = self.average_rating * self.count_of_ratings
    total_rating += taxa
    self.count_of_ratings += 1
    self.average_rating = round(total_rating / self.count_of_ratings, 1)
 ```